### PR TITLE
bearssl: init at 0.6 & s6-networking bearssl backend

### PR DESCRIPTION
**Pull request by Profpatsch**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/bearssl/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+stdenv.mkDerivation {
+  pname = "bearssl";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://www.bearssl.org/bearssl-${version}.tar.gz";
+    inherit sha256;
+  };
+
+  outputs = [ "bin" "lib" "dev" "out" ];
```
Comment (sternenseemann) [inline pkgs/development/libraries/bearssl/default.nix:17]: Wouldn't `lib` be a more appropriate default output?

File: pkgs/development/libraries/bearssl/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
+    install -D build/brssl $bin/brssl
+    install -D build/testcrypto $bin/testcrypto
+    install -Dm644 build/libbearssl.so $lib/lib/libbearssl.so
+    install -Dm644 build/libbearssl.a $lib/lib/libbearssl.a
+    install -Dm644 -t $dev/include inc/*.h
+    touch $out
```
Comment (sternenseemann) [inline pkgs/development/libraries/bearssl/default.nix:27]: Isn't `$out` a directory propagating `lib` and `bin`?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117136` at 1f1c0a61 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/46a01c38c4701122fadb908ccbb0e1f3">bearssl</a></li></li>
    <li><a href="https://gist.github.com/94e12002923b8918606f9e870635fe46">s6-networking</a></li> (skawarePackages.s6-networking)</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/libraries/bearssl/default.nix:21:3:
```
   |
21 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/bearssl/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  '';
```
Comment (sternenseemann) [inline pkgs/development/libraries/bearssl/default.nix:28]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -D build/brssl $bin/brssl
    install -D build/testcrypto $bin/testcrypto
    install -Dm644 build/libbearssl.so $lib/lib/libbearssl.so
    install -Dm644 build/libbearssl.a $lib/lib/libbearssl.a
    install -Dm644 -t $dev/include inc/*.h
    touch $out
    runHook postInstall
  '';
```

File: pkgs/development/libraries/bearssl/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        suites and extensions; BearSSL should implement most of them, and
+        allow extra algorithm implementations to be added afterwards,
+        possibly from third parties.
+    '';
+    license = lib.licenses.mit;
+    platforms = lib.platforms.all;
+    maintainers = [ lib.maintainers.Profpatsch ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/bearssl/default.nix:59]:
```suggestion
  };
}
```

File: pkgs/development/libraries/bearssl/default.nix [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  installPhase = ''
+    install -D build/brssl $bin/brssl
+    install -D build/testcrypto $bin/testcrypto
+    install -Dm644 build/libbearssl.so $lib/lib/libbearssl.so
+    install -Dm644 build/libbearssl.a $lib/lib/libbearssl.a
+    install -Dm644 -t $dev/include inc/*.h
+    touch $out
+  '';
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/bearssl/default.nix:30]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/bearssl/default.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        special contexts like bootstrap code.
+
+      * Be feature-rich and extensible. SSL/TLS has many defined cipher
+        suites and extensions; BearSSL should implement most of them, and
+        allow extra algorithm implementations to be added afterwards,
+        possibly from third parties.
+    '';
+    license = lib.licenses.mit;
+    platforms = lib.platforms.all;
+    maintainers = [ lib.maintainers.Profpatsch ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/bearssl/default.nix:56]:
```suggestion
    license = licenses.mit;
    platforms = platforms.all;
    maintainers = [ maintainers.Profpatsch ];
```

File: pkgs/development/libraries/bearssl/default.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+in
+stdenv.mkDerivation {
+  pname = "bearssl";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://www.bearssl.org/bearssl-${version}.tar.gz";
+    inherit sha256;
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/bearssl/default.nix:15]:
```suggestion
stdenv.mkDerivation rec {
  pname = "bearssl";
  version = "0.6";

  src = fetchurl {
    url = "https://www.bearssl.org/bearssl-${version}.tar.gz";
    sha256 = "057zhgy9w4y8z2996r0pq5k2k39lpvmmvz4df8db8qa9f6hvn1b7";
  };
```

File: pkgs/development/libraries/bearssl/default.nix (hunk 1 above)
Comment (Profpatsch) [reply pkgs/development/libraries/bearssl/default.nix:17]: `bin` is always expected to come first, the code that needs lib usually knows to default to it.

File: pkgs/development/libraries/bearssl/default.nix (hunk 2 above)
Comment (Profpatsch) [reply pkgs/development/libraries/bearssl/default.nix:27]: It didn’t work without the touch …

File: pkgs/development/libraries/bearssl/default.nix (hunk 4 above)
Comment (Profpatsch) [reply pkgs/development/libraries/bearssl/default.nix:59]: Sorry not doing whitespace changes

File: pkgs/development/libraries/bearssl/default.nix (hunk 3 above)
Comment (Profpatsch) [reply pkgs/development/libraries/bearssl/default.nix:28]: done

File: pkgs/development/libraries/bearssl/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/bearssl/default.nix:61]: Should I do them for you after this is merged?

File: pkgs/development/libraries/bearssl/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/libraries/bearssl/default.nix:17]: Fair enough :)

**Pull request merged**